wrappers/src/dictionary_cs.cpp

```diff
@@ -155,11 +155,24 @@ extern "C" {
         delete dictionary;
     }
 
```

Comment:
Need help to find a good place to put a util function like `get_key_path_array()` that applies for all these cases

Reply:
Yep. Probably you can generalise this method by having it accept the schema as input